runtime/sema/checker.go

```diff
@@ -1762,8 +1762,8 @@ func (checker *Checker) checkCharacterLiteral(expression *ast.StringExpression)
 	)
 }
 
-func (checker *Checker) isReadableAccess(access ast.Access) bool {
-	switch checker.Config.AccessCheckMode {
+func (mode AccessCheckMode) IsReadableAccess(access ast.Access) bool {
```

Comment:
Good idea to decouple this from `Checker` by changing the receiver. Please move it to `AccessCheckMode` (`accesscheckmode.go`). Same for `IsWriteableAccess` below.